slsim/Sources/supernovae.py

```diff
@@ -81,3 +103,39 @@ def get_apparent_magnitude(self, time, band, zpsys="AB"):
         """
 
         return self.bandmag(band, zpsys, time)
+
+    def set_magnitudes(
+        self,
+        absmag,
+        abs_mag_band,
+        peak_apparent_mag,
+        peak_apparent_mag_band,
+        magsys,
+        cosmo=cosmology.FlatLambdaCDM(H0=70, Om0=0.3),
+    ):
+        """Sets the amplitude of the source component of the model according to the
+        desired absolute magnitude in the specified band.
+
+        If the absolute magnitude is not given, then sets the amplitude of the source
+        component of the model according to a peak apparent magnitude.
+
+        If neither the absolute magnitude nor apparent magnitude are given, a warning
+        message is displayed.
+        """
+        if absmag is not None:
+            if abs_mag_band is None:
+                raise Exception(
```

Comment:
I think the lines of code in this definition can be tested as they do not require the import of the configuration files